doc/guides/tracing/index.rst

```diff
@@ -140,6 +140,24 @@ and exits (only distinguishes between idle thread, non idle thread and scheduler
 Enable this format with the :option:`CONFIG_TRACING_CPU_STATS` option.
 
 
+User-Defined Tracing
+====================
+
+This tracing format allows the user to define functions to perform any work desired
+when a task is switched in or out, when an interrupt is entered or exited, and when the cpu
+is idle.
+
```

Comment:
Can you please elaborate a bit on the usecase here and why you want to do that? Based on the above this seems like misuse of tracing to do something else not related to tracing, but maybe I am misunderstanding this.

Reply:
Let me know if the below is sufficient.  If so, I can add it:

Examples include
 - simple toggling of GPIO for external scope tracing while minimizing extra cpu load
 - generating/outputting trace data in a non-standard or proprietary format that can not be supported by the other tracing systems